src/sentry_core.c

```diff
@@ -231,6 +232,15 @@ sentry_transport_free(sentry_transport_t *transport)
     sentry_free(transport);
 }
 
+void
+sentry__enforce_disk_transport()
+{
+    // Freeing the old transport would, in the case of the curl transport, try
+    // to flush its send queue, which I’m not sure we can do in the signal
+    // handler. So rather we just leak it.
+    g_options->transport = sentry_new_disk_transport(g_options->database_path);
```

Comment:
So I was already thinking a bit about this, since the locking (or lack of it) around g_options really bothers me…
Apart from this very method (which we might solve differently anyway), the `sentry_options_t` is immutable anyway. So the lock is only really required to protect the global `g_options`.

I an imaginary future in which we might have multiple instances anyway, options would be thinging akin to `AtomicPtr<Arc<…>>`, `AtomicPtr` to work with the global instance, and `Arc` because we have multiple references, from worker threads etc… The only problem I see here is that we currently have a public `sentry_get_options` API. This would have to change in some way. Either require users to unref, or use a callback based API.